lib/spack/spack/build_systems/compiler.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import itertools
+import os
+import re
+import sys
+from typing import Sequence
+
+import llnl.util.tty as tty
+from llnl.util.lang import classproperty
+
+import spack.compiler
+import spack.package_base
+
+
+def true():
+    return True
+
+
+class CompilerPackage(spack.package_base.PackageBase):
+    """A Package mixin for all common logic for packages that implement compilers"""
```

Comment:
(nitpick) Maybe let's make it clear that right now this is about detection:
```suggestion
    """Mixin with common logic used when detecting compilers"""
```